src/iso_alloc_random.c

```diff
@@ -21,15 +21,24 @@
 #error "unknown OS"
 #endif
 
-
 /* Adapted from Daniel Lemire (@lemire). The code can be found at
  * https://github.com/lemire/testingRNG/blob/master/source/wyhash.h
- * This is adapted from wyhash from @wangyi-fudan */
+ * Which is adapted from wyhash from @wangyi-fudan
+ *
+ * This is not suitable for anything cryptographic. We use it in
+ * IsoAlloc for generating pointer masks, shuffling arrays etc,
+ * and the initial seed is refreshed with rand_uint64() each time
+ * we create a new zone.
+ *
+ * The magic values below are just random 64 bit primes. These values
+ * are combined with our seed and then run through the mum function.
+ * See the original wyhash paper for more details.
+ * https://github.com/wangyi-fudan/wyhash/ */
 INTERNAL_HIDDEN INLINE uint64_t us_rand_uint64(uint64_t *seed) {
```

Comment:
What does the `us_` stands for in `us_rand_uint64`? If it's `unsafe`, then it should be named `unsafe_rand_uint64` to make it obvious.

Reply:
Good question. It was supposed to be userspace because the old implementation was making a getrandom syscall but I don't like that shorthand either. Im open to renaming this to something better.